ravenframework/SupervisedLearning/DynamicModeDecompositionControl.py

```diff
@@ -245,19 +256,21 @@ def __evaluateLocal__(self,featureVals):
     ### Extract the initial state vector shape(n_requests,n_stateID)
     initStates = np.asarray([featureVals[:, :, self.features.index(par)] for par in self.initStateID]).T[0, :, :]
     # Initiate the evaluation array for evalX and evalY
-    evalX = np.zeros((len(indeces), tsEval, len(self.initStateID)))
-    evalY = np.zeros((len(indeces), tsEval, len(self.outputID)))
+    evalX = np.zeros((len(indices), tsEval, len(self.initStateID)))
+    evalY = np.zeros((len(indices), tsEval, len(self.outputID)))
 
-    for cnt, index in enumerate(indeces):
+    for cnt, index in enumerate(indices):
       # Centralize uVector and initState when required.
       if self.dmdParams['centerUXY']:
-        uVector = uVector - self.actuatorVals[0, index, :]
+        # use np.expand_dims to ensure this works with multiple actuators
+        uVector = uVector - np.expand_dims(self.actuatorVals[0, index, :], axis=tuple(range(1,len(uVector.shape))))
         initStates = initStates - self.stateVals[0, index, :]
       evalX[cnt, 0, :] = initStates
       evalY[cnt, 0, :] = np.dot(self.__Ctilde[index, :, :], evalX[cnt, 0, :])
       ### perform the self-propagation of X, X[k+1] = A*X[k] + B*U[k] ###
       for i in range(tsEval-1):
-        xPred = np.reshape(self.__Atilde[index,:,:].dot(evalX[cnt,i,:]) + self.__Btilde[index,:,:].dot(uVector[cnt,:,i]),(-1,1)).T
+        # make sure that Btilde dot uVector works correctly for multiple inputs and has the same shape as aTilde dot evalX
+        xPred = np.reshape(self.__Atilde[index,:,:].dot(evalX[cnt,i,:]) + (self.__Btilde[index,:,:].dot(uVector[:,:,i])).reshape(-1,), (-1,1)).T
```

Comment:
I'm not sure why 'cnt' index is not present in uVector anymore. I would like @wanghy-anl  to take a look at this PR.

Reply:
Including the 'cnt' index in uVector results in cycling through each actuator individually, B*U for B (n x m) and U (m x p) results in B(n x m)*U(1 x p). For only one input u, there is no error, but as soon as you add another input the error pops up.

Reply:
> Including the 'cnt' index in uVector results in cycling through each actuator individually, B*U for B (n x m) and U (m x p) results in B(n x m)*U(1 x p). For only one input u, there is no error, but as soon as you add another input the error pops up.

I have some reservations here. In alfoa/feature_selection branch we implemented it differently. Let me do some test on my side and get back to you.

Reply:
@dgarrett622 @wangcj05 This removal of `cnt` in `uVector` is OK. `cnt` always have a value of zero during the evaluation step, and like Daniel explained, we need to do the dot product for all the m variables in `uVector[:,:,i]`. I tested the newly added line #273 in alfoa/feature_selection branch and it returned the same result.